ipn/prefs.go

```diff
@@ -851,6 +851,9 @@ func exitNodeIPOfArg(s string, st *ipnstate.Status) (ip netip.Addr, err error) {
 		// address specified is valid.
 		if st.BackendState == "Running" {
 			ps, ok := peerWithTailscaleIP(st, ip)
+			if !ok && slices.Contains(st.TailscaleIPs, ip) {
+				return ip, errors.New("cannot use self as exit node")
```

Comment:
There's a very similar check below on line 864 - perhaps we should just do it before we check `BackendState`?

Reply:
Good suggestion, done.